dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -105,31 +100,48 @@ export default defineComponent({
      * Creates kpi data objects to pass them to the data table.
      * @param kpiKey The field name of a kpi
      * @param kpiValue The corresponding value to the kpiKey
-     * @param subAreaKey The sub area to which the kpi belongs
+     * @param subcategory The sub category to which the kpi belongs
      * @param dataIdOfLksgDataset The value of the date kpi of an LkSG dataset
      */
     createKpiDataObjects(
       kpiKey: string,
-      kpiValue: object | string | number,
-      subAreaKey: string,
+      kpiValue: object | string | Array<string> | number | null,
+      subcategory: Subcategory,
       dataIdOfLksgDataset: string
     ): void {
+      const kpi = subcategory.fields.filter((field) => field.name === kpiKey)[0];
       if (kpiKey === "totalRevenue" && typeof kpiValue === "number") {
         kpiValue = this.convertToMillions(kpiValue);
       }
+      if (kpiKey === "industry" || kpiKey === "subcontractingCompaniesIndustries") {
+        kpiValue = Array.isArray(kpiValue)
+          ? kpiValue.map((naceCodeShort: string) => naceCodeMap.get(naceCodeShort)?.label ?? naceCodeShort)
+          : naceCodeMap.get(kpiValue as string)?.label ?? kpiValue;
+      }
+      if (kpiKey === "subcontractingCompaniesCountries") {
+        kpiValue = Array.isArray(kpiValue)
+          ? kpiValue.map(
+              (countryCodeShort: string) => getCountryNameFromCountryCode(countryCodeShort) ?? countryCodeShort
+            )
+          : getCountryNameFromCountryCode(kpiValue as string) ?? kpiValue;
+      }
+      kpiValue = kpi.options?.filter((option) => option.value === kpiValue)[0]?.label ?? kpiValue;
       let indexOfExistingItem = -1;
-      const kpiDataObject = {
-        subAreaKey: subAreaKey == "general" ? `_${subAreaKey}` : subAreaKey,
+      const kpiData = {
+        subcategoryKey: subcategory.name == "masterData" ? `_${subcategory.name}` : subcategory.name,
+        subcategoryLabel: subcategory.label ? subcategory.label : subcategory.name,
         kpiKey: kpiKey,
+        kpiLabel: kpi?.label ? kpi.label : kpiKey,
+        kpiDescription: kpi?.description ? kpi.description : "",
         [dataIdOfLksgDataset]: kpiValue,
-      };
+      } as kpiDataObject;
       indexOfExistingItem = this.kpiDataObjects.findIndex(
```

Comment:
can't kpiDataObjects be a map key->kpiDataObject? That would avoid the tedious index handling

Reply:
yes, done

Reply:
Changed to Map<kpiKey, kpiDataObject>. The set and get functionalities make the Map indeed very useful here.

Reply:
ok